### ISSUE TITLE
LONGCHAR return value from method should be assigned to longchar variable

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#838]
It is possible to save the return value from a method that returns LONGCHAR into a character variable.
This produces obvious problems and a rule to catch this would be nice.

```
DEFINE VARIABLE lcFoo AS CHARACTER NO-UNDO.
lcFoo = Foo:MethodThatReturnsLongChar().
```

**Comment by stefandrissen** [comment]
This also applies to user defined functions that return a longchar. 

When the return value is passed through the ``substring`` functions the rule should not be triggered if the length is ok:
````
def var lcfoo as char.
def var iguess_my_value as int.

lcfoo = string( foo:mylongchar() ). // <- not ok
lcfoo = substring( foo:mylongchar(), 1, 30000 ). // <- ok
lcfoo = substring( foo:mylongchar(), 1, 40000 ). // <- not ok
lcfoo = substring( foo:mylongchar(), 1, iguess_my_value ). // <- ???
````

**Comment by gquerret** [comment]
Definitely worth a rule.

**Comment by PeterJudge-PSC** [comment]
And for `INT64` => `INTEGER`  too?

**Comment by gquerret** [comment]
> And for `INT64` => `INTEGER` too?

Could be done too. I think that it was already requested, and an issue is already there in GitHub

**Comment by movedoa** [comment]
> > And for `INT64` => `INTEGER` too?
> 
> Could be done too. I think that it was already requested, and an issue is already there in GitHub

Its #701

**Comment by gquerret** [comment]
Rule VariableOverflow will detect those patterns:
 * assignment of INT64 to INT
 * assignment of LONGCHAR / CLOB to CHAR
 * INT function call on INT64 (#701)
 * STRING function call on LONGCHAR (#508)
I'll probably merge the SubstituteLongChar in this rule as well.

**Comment by gquerret** [comment]
Added in develop. Note that the expression engine relies on rcode for methods return type.
